image/image.go

```diff
@@ -312,18 +329,33 @@ func bootstrapToRootDir(sto Store, model *asserts.Model, opts *Options, local *l
 		}
 
 		seen[name] = true
+		typ := info.Type
 
 		// if it comes from the store fetch the snap assertions too
 		// TODO: support somehow including available assertions
 		// also for local snaps
 		if info.SnapID != "" {
-			err = FetchAndCheckSnapAssertions(fn, info, f, db)
+			snapDecl, err := FetchAndCheckSnapAssertions(fn, info, f, db)
 			if err != nil {
 				return err
 			}
+			var kind string
+			switch typ {
+			case snap.TypeKernel:
+				kind = "kernel"
+			case snap.TypeGadget:
+				kind = "gadget"
+			}
+			if kind != "" { // kernel or gadget
+				publisher := snapDecl.PublisherID()
+				if publisher != model.BrandID() && publisher != "canonical" {
```

Comment:
if the logic gets more complicated we could (I can add a question TODO?) , atm is a 3 lines check (and we would need a place to put it)